api/v1alpha1/dnsrecord_types.go

```diff
@@ -128,6 +139,20 @@ func (s *DNSRecord) GetRootDomain() (string, error) {
 	return domain, nil
 }
 
+func (s *DNSRecord) Validate() error {
+	if s.Spec.RootHost != nil {
+		if len(strings.Split(*s.Spec.RootHost, ".")) <= 1 {
+			return fmt.Errorf("invalid domain format no tld discovered")
+		}
+		for _, ep := range s.Spec.Endpoints {
+			if !strings.HasSuffix(ep.DNSName, *s.Spec.RootHost) {
+				return fmt.Errorf("invalid endpoint discovered %s all endpoints should be equal to or end with the rootHost %s", ep.DNSName, *s.Spec.RootHost)
+			}
+		}
+	}
+	return nil
+}
```

Comment:
Should we also consider using CEL validations for some of these where possible? I know other kuadrant components were leaning more toward using those. Probably not possible here, just mentioning it.

Reply:
k also need to deal with a wildcard

Reply:
I guess no endpoints is also an error

Reply:
```
apiVersion: kuadrant.io/v1alpha1
kind: DNSRecord
metadata:
  labels:
    app.kubernetes.io/name: dnsrecord
    app.kubernetes.io/instance: dnsrecord-sample
    app.kubernetes.io/part-of: dns-operator
    app.kubernetes.io/managed-by: kustomize
    app.kubernetes.io/created-by: dns-operator
  name: dnsrecord-sample
spec:
  managedZone:
    name: dev-mz-aws
  rootHost: '*.cb.hcpapps.net'
  endpoints:
    - dnsName: simple.cb.hcpapps.net
      recordTTL: 60
      recordType: A
      targets:
        - 52.215.108.61
        - 52.30.101.221
    - dnsName: not-simple.cb.hcpapps.net
      recordTTL: 60
      recordType: A
      targets:
        - 52.215.108.61
        - 52.30.101.221     
  ```
Like this would be ok